src/human_encoding/named_node.rs

```diff
@@ -33,7 +33,7 @@ pub struct Named<N> {
 impl<J: Jet> node::Marker for Named<Commit<J>> {
     type CachedData = NamedCommitData<J>;
     type Witness = <Commit<J> as node::Marker>::Witness;
-    type Disconnect = <Commit<J> as node::Marker>::Disconnect;
+    type Disconnect = Arc<str>;
     type SharingId = Arc<str>;
```

Comment:
in 428fdf3628534e4677e31f9f37075f446901b4ef:

Can wait for a future PR but these `Arc<str>`s should both be newtypes I think.

Reply:
We can introduce the newtype `HoleName` and use it in `WitnessOrHole` and in `NamedCommitNode::Disconnect`. `Arc<str>` is used in 428fdf3628534e4677e31f9f37075f446901b4ef only for hole names, unless I am missing something.

Reply:
Yep, I believe you introduce and use it only for hole names.

We should make a second newtype for the `SharingId`. That way we can't get them confused when we're threading all this stuff through the abstract node infrastructure.